src/main/scala/is/hail/table/Table.scala

```diff
@@ -799,7 +799,7 @@ class Table(val hc: HailContext,
 
   def flatten(): Table = {
     val localSignature = signature
-    val keySignature = TStruct(keyFields)
+    val keySignature = TStruct(keyFields.map { f => f.name -> f.typ}: _*)
```

Comment:
missing space after `f.typ`. But I think you should just use parentheses here, not braces.